language/evm/move-to-yul/src/solidity_ty.rs

```diff
@@ -0,0 +1,627 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Representation of solidity types and related functions.
+//! TODO: struct and function type
+
+use anyhow::{anyhow, Context as AnyhowContext};
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::{fmt, fmt::Formatter};
+
+use move_model::{
+    model::{FunctionEnv, QualifiedId, StructId},
+    ty::{PrimitiveType, Type},
+};
+
+use crate::context::Context;
+
+// Solidity signature matching
+static SIG_REG: Lazy<Regex> = Lazy::new(|| {
```

Comment:
Just a note that you can move the static declaration actually inside the function where you need it, if you want so. This makes it definitely simpler during development as those things are closer together, but also easier to read.